inc/Engine/Common/AbstractFileSystem.php

```diff
@@ -0,0 +1,97 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Common;
+
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use WP_Filesystem_Direct;
+
+abstract class AbstractFileSystem {
+	/**
+	 * WP Filesystem instance.
+	 *
+	 * @var WP_Filesystem_Direct
+	 */
+	protected $filesystem;
+
+	/**
+	 * Constructor method.
+	 * Initializes a new instance of the Controller class.
+	 *
+	 * @param WP_Filesystem_Direct $filesystem Filesystem class.
+	 */
+	public function __construct( $filesystem = null ) {
+		$this->filesystem = $filesystem ?? rocket_direct_filesystem();
+	}
+
+	/**
+	 * Write to file.
+	 *
+	 * @param string $file_path File path to store the file.
+	 * @param string $content   File content(data).
+	 *
+	 * @return bool
+	 */
+	protected function write_file( string $file_path, string $content ): bool {
+		return $this->filesystem->put_contents( $file_path, $content, rocket_get_filesystem_perms( 'file' ) );
+	}
+
+	/**
+	 * Get the content of a file
+	 *
+	 * @param string $file The file content to get.
+	 *
+	 * @return string
+	 */
+	protected function get_file( string $file ): string {
```

Comment:
I think the naming of the method should be different, we are getting the content of the file, not the file itself